File: 0049-group-anagrams/0049-group-anagrams.py
# from collections import defaultdict

class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        # groups = {}
        groups = defaultdict(list)

        for i in strs:
            key = tuple(sorted(i))
            # if key not in groups:
            #     groups[key] = []
            groups[key].append(i)

        return groups.values()

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> groups;

        for (int i = 0; i < strs.size(); i++){
            string s = strs[i];
            sort(s.begin(), s.end());
            groups[s].push_back(strs[i]);
        }

        vector<vector<string>> ans;

        for (auto& group : groups) {
            ans.push_back(group.second);
        }

        return ans;
    }
};

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna